src/components/structures/MessagePanel.tsx

```diff
@@ -405,11 +410,11 @@ export default class MessagePanel extends React.Component<IProps, IState> {
             return false; // ignored = no show (only happens if the ignore happens after an event was received)
         }
 
-        if (this.showHiddenEventsInTimeline) {
+        if (this.context?.showHiddenEventsInTimeline ?? this.showHiddenEventsInTimeline) {
             return true;
         }
 
-        if (!haveTileForEvent(mxEv)) {
+        if (!haveTileForEvent(mxEv, this.context?.showHiddenEventsInTimeline)) {
```

Comment:
Surely we should either read this in both or neither?
Given we already paid the cost of reading the SettingsStore is there any benefit?

Reply:
I'm not sure what you mean by "read this in both or neither"

Reply:
Hopefully suggestion above clears things up - we're sometimes looking at `this.context?.showHiddenEventsInTimeline ?? this.showHiddenEventsInTimeline` sometimes just the one in context, we should be consistent.

Reply:
Ah, well the issue is that `this.showHiddenEventsInTimeline` is only set once, in the constructor (as it should to avoid spamming the settings store), so if we just used that everywhere, things wouldn't react properly to context changes. But sure, I can at least be consistent about using `this.context?.showHiddenEventsInTimeline ?? this.showHiddenEventsInTimeline` everywhere.